crowdsourcehinter/static/js/src/crowdsourcehinter.js

```diff
@@ -0,0 +1,305 @@
+function CrowdsourceHinter(runtime, element, data){
+
+    var executeHinter = true;
+    var isShowingHintFeedback = false;
+    var voted = false;
+    var correctSubmission = false;
+    
+    $(".crowdsourcehinter_block", element).hide();
+
+    if(!executeHinter){
+        return;
+    }
+
+    /**
+     * Set executeHinter to false, disabling the hinter xblock. Triggered by switching units
+     * in edX course.
+     * This is a workaround for when a student switches to/from a unit and causes multiple
+     * instances of the hinter to be running.
+     */
+    function stopScript(){
+        executeHinter = false;
+    }
+    Logger.listen('seq_next', null, stopScript);
+    Logger.listen('seq_prev', null, stopScript);
+    Logger.listen('seq_goto', null, stopScript);
+
+    /**
+     * Get a hint to show to the student after incorrectly answering a question.
+     * @param data is data generated by the problem_graded event
+     */
+    function get_hint(data){
+        $(".crowdsourcehinter_block", element).show();
+        $.ajax({
+            type: "POST",
+            url: runtime.handlerUrl(element, 'get_hint'),
+            data: JSON.stringify({"submittedanswer": unescape(data[0])}),
+            success: showHint
+        });
+    }
+
+    /**
+     * Start student hint feedback. This function is called after the student answers
```

Comment:
Rename **feedback** globally. Feedback is too generic. 

* hints are *feedback* to the student. 
* voting/contributing/editing are *feedback* from the student on the system. 

Search for feedback, and write something unambiguous everywhere. Do not use the term anywhere in the code.